components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java

```
@@ -187,6 +207,14 @@ public Response doPublish(@PathParam("type") String type) {
                         registry.put(DEFAULT_RESOURCE_LOCATION + type + ".exist", resource);
                     }
                 }
+
+                if(!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                    publishDynamicEventReceivers(type, tenantDomain);
+                }
+                publishDynamicEventReceivers(type,MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+                publishDynamicEventStream(type,MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
```

Better to return proper HTTP Error Code than suppressing the exception.